src/app/public/src/modules/edit-button/edit-button.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { StacheJsonDataService } from '../shared';
+import { StacheNavService } from '../nav';
+
+@Component({
+  selector: 'stache-edit-button',
+  templateUrl: './edit-button.component.html',
+  styleUrls: ['./edit-button.component.scss']
+})
+export class StacheEditButtonComponent implements OnInit {
+  @Input()
```

Comment:
When would we ever want to change this text to something other than 'Edit' ?  As this is a simple button component that builds it's own url off the config, I don't see this being manually added anywhere that would need alterable text.

Reply:
I could see public-facing SPAs being particular about what language they use for the Edit button. I think we can enforce it internally, but it's naive to think no one will ever want to customize the name of the button.

Reply:
I think we should be consistent with the language we use for this component across all SPAs that want to make use of it. With that being an input, in order for consumers to make use of this button, they wouldn't be able to globally turn it on.  They would have to manually add the button with that input name on *every page* they wanted it to display.  

With how this component takes care of all the other stuff for you (formatting the URLs etc.) Not being prescriptive and deliberate with the language we use seems out of place, especially with how difficult it would be to circumnavigate.

I would err on the side of maybe changing the text from `Edit` to `Contribute` or something possibly, but I would not leave it up to individual SPAs to determine their best word usage here.

Reply:
I could see making this a global config setting as well.

Reply:
I can see the point of maintaining consistency between SPAs, but I don't think it's a huge deal to leave it up to the consumer to make this say whatever they want, "Edit on GitHub", "Edit on VSTS", etc.